src/Console/Application.php

```diff
@@ -286,10 +290,19 @@ public function buildContainer()
         };
 
         // VIOLATION RENDERER
-        $c['violation.renderer'] = function ($c) {
+        $c['violation.renderer.console'] = function ($c) {
             return new Renderer\ConsoleOutputRenderer(new ConsoleOutput(), $c['violation.message_helper']);
         };
 
+        $c['violation.renderer.html'] = function ($c) {
+            return new Renderer\HtmlOutputRenderer(
+                new ConsoleOutput(),
+                $c['violation.message_helper'],
+                new Filesystem(),
+                './output.html'
```

Comment:
Right, I did not notice the interface. My bad.

Right now, this interface does not have a purpose. Whenever one of the renderer is called, we know exactly which one it is. So, we could drop the interface. But that's obviously not the solution, it would just reflect the current implementation of the command.

Instead, we should operate on an abstract collection of renderers and call the `renderViolations` method on each. That of course means, that our renderers need a state again.

What you could do is implement a `HtmlOutputRendererFactory` and put it into the container, like this:

```php
class HtmlOutputRendererFactory {
    private $messageHelper;
    private $filesystem;

    public function __construct(MessageHelper $messageHelper, Filesystem $filesystem)
    {
        $this->messageHelper = $messageHelper;
        $this->filesystem = $filesystem;
    }

    public function createhtmlOutputRenderer($outputFile)
    {
        return new HtmlOutputRenderer($this->messageHelper, $this->filesystem, $outputFile);
    }
}
```

The factory itself is stateless, but it can create a configured renderer for you.